Update Get-CsOnlineUser.md - Additional Changes (#9019)
MicrosoftDocs/office-docs-powershell: skype/skype-ps/skype/Get-CsOnlineUser.md

The following updates are applicable for TeamsOnly customers using Microsoft Teams PowerShell version 3.0.0 and above.

> *New User Attributes*

Reviewer: ```suggestion
*New User Attributes*
```
```suggestion
*New user attributes*
```

FeatureTypes – Array of unique strings specifying what features are enabled for a user (plan not displayed). This attribute is an alternatitive to several attributes that have been dropped as outlined in the next section.
Some of the commonly used FeatureTypes include:
- AudioConferencing
- PhoneSystem
- CallingPlan

> *Dropped Attributes*  

Reviewer: ```suggestion
*Dropped attributes*  
```

- CountryAbbreviation
- UsageLocation

*Attributes renamed*

- FirstName renamed to GivenName
- DirSyncEnabled renamed to UserDirSyncEnabled
- MCOValidationErrors renamed to UserValidationErrors
- Enabled renamed to IsSipEnabled
- TeamsBranchSurvivabilityPolicy renamed to TeamsSurvivableBranchAppliancePolicy

*Attributes that have changed in meaning/format*

OnPremLineURI: This attribute is used to refer to both:
1. Previous LineURI set via OnPrem AD.
2. Previous Direct Routing numbers assigned to users via Set-CsUser.

> In Teams PowerShell Modules 3.0.0 and above OnPremLineURI will only refer to the LineURI set via OnPrem AD. Direct Routing numbers can be distinguished from Calling Plan users by looking at the FeatureTypes attribute.

Reviewer 2: nit: did you mean 'for' calling plan users?

Author: No I meant we can distinguish between Direct Routing numbers and Calling Plan using FeatureTypes

- CountryAbbreviation
- UsageLocation

> *Attributes renamed:*

Reviewer: ```suggestion
*Attributes renamed*
```

- FirstName renamed to GivenName
- DirSyncEnabled renamed to UserDirSyncEnabled
- MCOValidationErrors renamed to UserValidationErrors
- Enabled renamed to IsSipEnabled
- TeamsBranchSurvivabilityPolicy renamed to TeamsSurvivableBranchAppliancePolicy

*Attributes that have changed in meaning/format*

> OnPremLineURI: This attribute used to previously refer to both 

Reviewer: ```suggestion
OnPremLineURI: This attribute is used to previously refer to both 
```

>   - LineURI set via OnPrem AD
>   - Direct Routing numbers assigned to users via Set-CsUser

Reviewer: ```suggestion
  - LineURI set via OnPrem AD.
  - Direct Routing numbers assigned to users via Set-CsUser.
```

> In Teams PowerShell Modules 3.0.0 and above OnPremLineURI will only refer to LineURI set via OnPrem AD. Direct Routing numbers can be distinguised from Calling Plan users by looking at the FeatureTypes attribute.

Reviewer: ```suggestion
In Teams PowerShell Modules 3.0.0 and above OnPremLineURI will only refer to the LineURI set via OnPrem AD. Direct Routing numbers can be distinguished from Calling Plan users by looking at the FeatureTypes attribute.
```

> - The output format of AssignedPlan and ProvisionedPlans have now changed from xml to json
> - The output format of Policies has now changed from String to json

Reviewer: ```suggestion
- The output format of AssignedPlan and ProvisionedPlans have now changed from XML to JSON.
- The output format of Policies has now changed from String to JSON.
```

*Changes in "-Filter" parameter*

> - Assigned Plan filter: Previous format will no longer be supported. Existing filters like `AssignedPlan -eq '<some-xml-string>'` will stop working. This will need to be modified to one of the following formats:

Reviewer: ```suggestion
- Assigned Plan filter: The previous format will no longer be supported. Existing filters like `AssignedPlan -eq '<some-xml-string>'` will stop working. This needs to be modified to one of the following formats:
```

>   - ``AssignedPlan -eq "*MCO" or 'MCO*' or '*MCO*' - for contains checks``
>   - ``The wildcard operator '*' will only work with AssignedPlan when used in conjunction with the '-eq' operator``

Reviewer: ```suggestion
  - ``AssignedPlan -eq "*MCO" or 'MCO*' or '*MCO*' - for contains checks``.
  - ``The wildcard operator '*' will only work with AssignedPlan when used in conjunction with the '-eq' operator``.
```

- EnterpriseVoiceEnabled filter

>   - The new format is as follows: ``EnterpriseVoiceEnabled -eq $true / $false``

Reviewer: ```suggestion
  - The new format is as follows: ``EnterpriseVoiceEnabled -eq $true / $false``.
```

> *Dropped input parameters:*

Reviewer: ```suggestion
*Dropped input parameters*
```

The following input paramters have beed dropped because they are no longer relevant to Teams:
```
[-LdapFilter <String>] 
[-OnOfficeCommunicationServer] 
[-OnModernServer]
[-UnassignedUser] 
[-OU <OUIdParameter>] 
[-DomainController <Fqdn>]
```

> *Dropped Filter Operators*

Reviewer: ```suggestion
*Dropped Filter operators*
```

> The following filter syntaxs are not supported in TeamsPowerShell Moduled 3.0.0 and above:

Reviewer: ```suggestion
The following filter syntaxes are not supported in TeamsPowerShell Moduled 3.0.0 and above:
```

- CountryAbbreviation
- UsageLocation

*Attributes renamed*

- FirstName renamed to GivenName
- DirSyncEnabled renamed to UserDirSyncEnabled
- MCOValidationErrors renamed to UserValidationErrors
- Enabled renamed to IsSipEnabled
- TeamsBranchSurvivabilityPolicy renamed to TeamsSurvivableBranchAppliancePolicy

*Attributes that have changed in meaning/format*

> OnPremLineURI: This attribute is used to previously refer to both:
>   - LineURI set via OnPrem AD.
>   - Direct Routing numbers assigned to users via Set-CsUser.

Reviewer: ```suggestion
OnPremLineURI: This attribute is used to refer to both:
  - Previous LineURI set via OnPrem AD.
  - Previous Direct Routing numbers assigned to users via Set-CsUser.
```

- The output format of AssignedPlan and ProvisionedPlans have now changed from XML to JSON.
- The output format of Policies has now changed from String to JSON.

*Changes in "-Filter" parameter*

- Assigned Plan filter: The previous format will no longer be supported. Existing filters like `AssignedPlan -eq '<some-xml-string>'` will stop working. This needs to be modified to one of the following formats:
  - ``AssignedPlan -eq "*MCO" or 'MCO*' or '*MCO*' - for contains checks``.
  - ``The wildcard operator '*' will only work with AssignedPlan when used in conjunction with the '-eq' operator``.

- EnterpriseVoiceEnabled filter
  - The new format is as follows: ``EnterpriseVoiceEnabled -eq $true / $false``.

*Dropped input parameters*

The following input paramters have beed dropped because they are no longer relevant to Teams:

```
[-LdapFilter <String>] 
[-OnOfficeCommunicationServer] 
[-OnModernServer]
[-UnassignedUser] 
[-OU <OUIdParameter>] 
[-DomainController <Fqdn>]
```
*Dropped Filter operators*
  

> The following filter syntaxes are not supported in TeamsPowerShell Moduled 3.0.0 and above:

Reviewer: ```suggestion
The following filter syntaxes are not supported in Teams PowerShell Moduled 3.0.0 and above:
```

For example, a filter that returns only users who have been enabled for Enterprise Voice would look like this, with EnterpriseVoiceEnabled representing the Active Directory attribute, -eq representing the comparison operator (equal to), and $True (a built-in Windows PowerShell variable) representing the filter value:

`{EnterpriseVoiceEnabled -eq $True}`
  
  *Changes in "-Filter" parameter*
- Assigned Plan filter - Previous format will no longer be supported. Existing filters like `AssignedPlan -eq '<some-xml-string>'` will stop working. This will need to be modified to one of the following formats:
  - ``AssignedPlan -eq "*MCO" or 'MCO*' or '*MCO*' - for contains checks``

> *Note:* The output format has also changed, the xml object is now a json object.

Reviewer: ```suggestion
*Note:* The output format has also changed, the XML object is now a JSON object.
```

- EnterpriseVoiceEnabled filter

>   - The new format is as follows: ``EnterpriseVoiceEnabled -eq $true / $false``

Reviewer: ```suggestion
  - The new format is as follows: ``EnterpriseVoiceEnabled -eq $true / $false``.
```

For example, a filter that returns only users who have been enabled for Enterprise Voice would look like this, with EnterpriseVoiceEnabled representing the Active Directory attribute, -eq representing the comparison operator (equal to), and $True (a built-in Windows PowerShell variable) representing the filter value:

`{EnterpriseVoiceEnabled -eq $True}`
  
  *Changes in "-Filter" parameter*
- Assigned Plan filter - Previous format will no longer be supported. Existing filters like `AssignedPlan -eq '<some-xml-string>'` will stop working. This will need to be modified to one of the following formats:
  - ``AssignedPlan -eq "*MCO" or 'MCO*' or '*MCO*' - for contains checks``

*Note:* The output format has also changed, the XML object is now a JSON object.

> 

Reviewer 2: Seems to be repeating, already saw it  on 273. Is this intended?

- TeamsCallingPolicy
- TeamsUpgradePolicy
- TeamsUpgradeOverridePolicy
- TeamsAppSetupPolicy
- TeamsAppPermissionPolicy
- TeamsVerticalPackagePolicy
- TeamsSurvivableBranchAppliancePolicy
- TeamsCallHoldPolicy
- TenantDialPlan
- OnlineVoicemailPolicy
- OnlineAudioConferencingRoutingPolicy
- TeamsAudioConferencingPolicy
- TeamsVdiPolicy
- TeamsFeedbackPolicy
- TeamsIPPhonePolicy
- TeamsShiftsAppPolicy
- TeamsShiftsPolicy
- TeamsTargetingPolicy
- TeamsTemplatePermissionPolicy
- TeamsSyntheticAutomatedCallPolicy
- TeamsMobilityPolicy
- TeamsCortanaPolicy
- TeamsMeetingBrandingPolicy
- TeamsNotificationAndFeedsPolicy
- TeamsVideoInteropServicePolicy
- TeamsEducationAssignmentsAppPolicy
- TeamsComplianceRecordingPolicy
- AssignedPlan
- EnterpriseVoiceEnabled
- Identity
- department
- UserDirSyncEnabled
- Title
- CountryAbbreviation
- UsageLocation
  

## OUTPUTS

*Attributes renamed:*

- FirstName renamed to GivenName
- DirSyncEnabled renamed to UserDirSyncEnabled
- MCOValidationErrors renamed to UserValidationErrors
- Enabled renamed to IsSipEnabled
- TeamsBranchSurvivabilityPolicy renamed to TeamsSurvivableBranchAppliancePolicy

*Attributes that have changed in meaning/format*

> OnPremLineURI: This attribute used to previously refer to both 

Reviewer: ```suggestion
OnPremLineURI: This attribute used to previously refer to both:
```

> OnPremLineURI: This attribute used to previously refer to both 
>   - LineURI set via OnPrem AD.
>   - Direct Routing numbers assigned to users via Set-CsUser.

Reviewer: ```suggestion
OnPremLineURI: This attribute used to refer to both:
  - Previous LineURI set via OnPrem AD.
  - Previous Direct Routing numbers assigned to users via Set-CsUser.
```

Author: @JohanFreelancer9 I had some additional changes which conflict with this commit. Please check now

> In Teams PowerShell Modules 3.0.0 and above OnPremLineURI will only refer to LineURI set via OnPrem AD. Direct Routing numbers can be distinguished from Calling Plan users by looking at the FeatureTypes attribute.

Reviewer: ```suggestion

In Teams PowerShell Modules 3.0.0 and above OnPremLineURI will only refer to LineURI set via OnPrem AD. Direct Routing numbers can be distinguished from Calling Plan users by looking at the FeatureTypes attribute.
```

- displayName
- givenName
- lineUri
- userPrincipalName
- ExternalAccessPolicy
- OnlineDialOutPolicy
- OnlineVoiceRoutingPolicy
- TeamsMeetingPolicy
- TeamsMeetingBroadcastPolicy
- TeamsMessagingPolicy
- TeamsCallParkPolicy
- TeamsEmergencyCallingPolicy
- TeamsEmergencyCallRoutingPolicy
- TeamsChannelsPolicy
- TeamsUpdateManagementPolicy
- TeamsCallingPolicy
- TeamsUpgradePolicy
- TeamsUpgradeOverridePolicy
- TeamsAppSetupPolicy
- TeamsAppPermissionPolicy
- TeamsVerticalPackagePolicy
- TeamsSurvivableBranchAppliancePolicy
- TeamsCallHoldPolicy
- TenantDialPlan
- OnlineVoicemailPolicy
- OnlineAudioConferencingRoutingPolicy
- TeamsAudioConferencingPolicy
- TeamsVdiPolicy
- TeamsFeedbackPolicy
- TeamsIPPhonePolicy
- TeamsShiftsAppPolicy
- TeamsShiftsPolicy
- TeamsTargetingPolicy
- TeamsTemplatePermissionPolicy
- TeamsSyntheticAutomatedCallPolicy
- TeamsMobilityPolicy
- TeamsCortanaPolicy
- TeamsMeetingBrandingPolicy
- TeamsNotificationAndFeedsPolicy
- TeamsVideoInteropServicePolicy
- TeamsEducationAssignmentsAppPolicy
- TeamsComplianceRecordingPolicy
- AssignedPlan
- EnterpriseVoiceEnabled
- Identity
- department
- UserDirSyncEnabled
- Title
- CountryAbbreviation
- UsageLocation
  

> *Dropped Filter Operators*
>   
> The following filter syntaxs are not supported in TeamsPowerShell Moduled 3.0.0 and above:

Reviewer: ```suggestion
*Dropped filter operators*
  
The following filter syntaxes are not supported in Teams PowerShell Moduled 3.0.0 and above:
```

```
-not (<simple/complex PS filter>)
<property> -like '*<text>'
<property> -like '*<text>*'
<property> -lt <value>
<property> -gt <value>
<PolicyPropertyName> -ge <value>, <PolicyPropertyName> -le <value>, <PolicyPropertyName> -gt <value>, <PolicyPropertyName> -lt <value>
```

## OUTPUTS

*Attributes renamed:*

- FirstName renamed to GivenName
- DirSyncEnabled renamed to UserDirSyncEnabled
- MCOValidationErrors renamed to UserValidationErrors
- Enabled renamed to IsSipEnabled
- TeamsBranchSurvivabilityPolicy renamed to TeamsSurvivableBranchAppliancePolicy

*Attributes that have changed in meaning/format*

> OnPremLineURI: This attribute used to previously refer to both 

Reviewer: ```suggestion
OnPremLineURI: This attribute is used to previously refer to both: 
```

>   - LineURI set via OnPrem AD
>   - Direct Routing numbers assigned to users via Set-CsUser

Reviewer: ```suggestion
  - LineURI set via OnPrem AD.
  - Direct Routing numbers assigned to users via Set-CsUser.
```

> In Teams PowerShell Modules 3.0.0 and above OnPremLineURI will only refer to LineURI set via OnPrem AD. Direct Routing numbers can be distinguised from Calling Plan users by looking at the FeatureTypes attribute.

Reviewer: ```suggestion
In Teams PowerShell Modules 3.0.0 and above OnPremLineURI will only refer to LineURI set via OnPrem AD. Direct Routing numbers can be distinguished from Calling Plan users by looking at the FeatureTypes attribute.
```

> - The output format of AssignedPlan and ProvisionedPlans have now changed from xml to json
> - The output format of Policies has now changed from String to json

Reviewer: ```suggestion
- The output format of AssignedPlan and ProvisionedPlans have now changed from XML to JSON.
- The output format of Policies has now changed from String to JSON.
```
